x-pack/plugins/entity_manager/server/lib/entities/ingest_pipeline/generate_history_processors.ts

```diff
@@ -140,52 +140,24 @@ export function generateHistoryProcessors(definition: EntityDefinition) {
       },
     },
     {
-      script: {
-        description: 'Generated the entity.id field',
-        source: cleanScript(`
-        // This function will recursively collect all the values of a HashMap of HashMaps
-        Collection collectValues(HashMap subject) {
-          Collection values = new ArrayList();
-          // Iterate through the values
-          for(Object value: subject.values()) {
-            // If the value is a HashMap, recurse
-            if (value instanceof HashMap) {
-              values.addAll(collectValues((HashMap) value));
-            } else {
-              values.add(String.valueOf(value));
-            }
-          }
-          return values;
-        }
-
-        // Create the string builder
-        StringBuilder entityId = new StringBuilder();
+      set: {
+        field: 'entity.id',
+        value: definition.identityFields
+          .map((identityField) => {
+            const field = `entity.identity.${identityField.field}`;
 
-        if (ctx["entity"]["identity"] != null) {
-          // Get the values as a collection
-          Collection values = collectValues(ctx["entity"]["identity"]);
+            if (!identityField.optional) {
+              return `{{{${field}}}}`;
+            }
 
-          // Convert to a list and sort
-          List sortedValues = new ArrayList(values);
-          Collections.sort(sortedValues);
+            const renderValue = `{{#${field}}}{{{${field}}}}{{/${field}}}`;
 
-          // Create comma delimited string
-          for(String instanceValue: sortedValues) {
-            entityId.append(instanceValue);
-            entityId.append(":");
-          }
+            const fallback = identityField.fallback ?? '?';
+            const renderFallback = `{{^${field}}}${fallback}{{/${field}}}`;
 
-            // Assign the entity.id
-          ctx["entity"]["id"] = entityId.length() > 0 ? entityId.substring(0, entityId.length() - 1) : "unknown";
-        }
-       `),
-      },
-    },
-    {
-      fingerprint: {
-        fields: ['entity.id'],
-        target_field: 'entity.id',
-        method: 'MurmurHash3',
+            return `${renderValue}${renderFallback}`;
```

Comment:
is there an easy way to unit test mustache template ?

Reply:
With the removal of optional fields I believe the template is so simple as to not justify a unit test.